metadata-ingestion/src/datahub/api/entities/datajob/datajob.py

```diff
@@ -103,6 +107,13 @@ def generate_tags_aspect(self) -> Iterable[GlobalTagsClass]:
     def generate_mcp(
         self, materialize_iolets: bool = True
     ) -> Iterable[MetadataChangeProposalWrapper]:
+        env: Optional[str] = None
+        if self.flow_urn.cluster in ALL_ENV_TYPES:
+            env = self.flow_urn.cluster
```

Comment:
for my understanding - with this change, env and cluster will be largely independent? and cluster will work like platform instance, except that it has an explicitly spot in the urn?

Reply:
We have some logic elsewhere that copies the env value into cluster

Can you give a one liner of exactly how cluster and env are supposed to be set? The fallbacks and validity checks make this pretty confusing

Reply:
`cluster` should be using valid values from FabricType only to make filters work in search.

Reply:
Can we enforce only valid FabricTypes and reject anything else?

Reply:
Did not reject everything else as that would be too disruptive to existing workflows. This is a compromise between making search filters work and not cause havoc. 

We have to change our recommendation and tell folks to only use the values from FabricType if they want filters to wrok.